### PR TITLE
[FIX] UI bugs

### PR DESCRIPTION
**Pull request by shivani170**
# Description
 This PR including bugs:

1.Template selection is not mandatory while creating an app, users can create an app with `Blank template`.

Screenshot 2021-05-28 at 2 39 19 PM
2. Currently, the env override in the `App Configuration> Environment override` tab accordion is empty when no cd pipeline is configured.

Screenshot 2021-05-28 at 2 44 54 PM

3. refactoring code of sso login

Fixes: https://github.com/devtron-labs/devtron/issues/285
Closes: https://github.com/devtron-labs/devtron/issues/378

### CODE REVIEW

File: src/components/app/details/appConfig/AppConfig.tsx [hunk 1]
```diff
@@ -372,13 +373,35 @@ function AppComposeRouter({ isUnlocked, navItems, respondOnSuccess, isCiPipeline
     </ErrorBoundary>
 }
 
+const EnvironmentOverrideDropdown = (environmentResult, environmentsLoading, url) => {
```
Comment (pghildiyal) [inline src/components/app/details/appConfig/AppConfig.tsx:376]: we should define type of arguments

Comment (shivani170) [reply src/components/app/details/appConfig/AppConfig.tsx:376]: fixing

File: src/components/login/SSOLogin.tsx [hunk 2]
```diff
@@ -337,78 +369,12 @@ export default class SSOLogin extends Component<SSOLoginProps, SSOLoginState> {
...
-                            <span className="tertiary-tab sso-icons">
-                                <aside className="login__icon-alignment"><img src={Openshift} /></aside>
-                                <aside className="login__text-alignment"> OpenShift</aside>
-                                <label>
-                                    {this.state.lastActiveSSO?.name == "openshift" ? <aside className="login__check-icon"><img src={Check} /></aside> : ''}
-                                </label>
-                            </span>
-                        </label>
-                    </div>
+                    <SSOLoginTab value={SSOProvider.google} SSOName="Google" checked={this.state.sso === SSOProvider.google} handleSSOClick={this.handleSSOClick} lastActiveSSO={this.state.lastActiveSSO?.name == SSOProvider.google} />
```
Comment (pghildiyal) [inline src/components/login/SSOLogin.tsx:372]: we can use for loop

File: src/components/app/create/CreateApp.tsx [hunk 3]
```diff
@@ -160,7 +160,7 @@ export class AddNewApp extends Component<AddNewAppProps, AddNewAppState> {
                             : null}
                     </span>
                     <span className="form__text-field-info">
-                        <Info className="form__icon form__icon--info" style={{height: "none"}} />
+                        <Info className="form__icon form__icon--info" style={{ height: "none" }} />
```
Comment (rashmirai21) [inline src/components/app/create/CreateApp.tsx:163]: height: "none" is an invalid style. remove it

[the remark this answers is not in the document]
Comment (shivani170) [reply src/components/app/create/CreateApp.tsx]: fixed

File: src/components/app/details/appDetails/AppMetrics.tsx [hunk 4]
```diff
@@ -138,14 +137,16 @@ export const AppMetrics: React.FC<{ appName: string, environment, podMap: Map<st
...
         if (!datasource.isHealthy) return;
-
+ 
         if (!isK8sVersionValid(k8sVersion)) {
             k8sVersion = DEFAULTK8SVERSION;
+          
             toast.warn(<div className="toast">
                 <div className="toast__title">Error Parsing K8sVersion</div>
-                <div className="toast__subtitle">Showing Graphs for {DEFAULTK8SVERSION} and above</div>
+                <div className="toast__subtitle">Showing Graphs for {k8sVersion} and above</div>
```
[the remark this answers is not in the document]
Comment (shivani170) [reply src/components/app/details/appDetails/AppMetrics.tsx:146]: fixed

File: src/components/login/SSOLogin.tsx [hunk 5]
```diff
@@ -36,6 +45,29 @@ const ssoMap = {
...
+    }
+}
+const SSOLoginTab: React.FC<{ handleSSOClick: (e) => void, checked: boolean, lastActiveSSO, value: string, SSOName: string }> = ({ handleSSOClick, checked, lastActiveSSO, value, SSOName }) => {
+    return <label className="tertiary-tab__radio">
+        <input type="radio" value={value} checked={checked} name="status" onClick={handleSSOClick} />
+        <span className="tertiary-tab sso-icons">
+            <aside className="login__icon-alignment"><SSOTabIcons SSOName={SSOName} /></aside>
+            <aside className="login__text-alignment">{SSOName}</aside>
+            <label>
+                {lastActiveSSO?.name == { value } ? <aside className="login__check-icon"><img src={Check} /></aside> : ""}
```
[the remark this answers is not in the document]
Comment (shivani170) [reply src/components/login/SSOLogin.tsx]: fixed

File: src/components/app/details/appConfig/AppConfig.tsx [hunk 6]
```diff
@@ -372,13 +373,35 @@ function AppComposeRouter({ isUnlocked, navItems, respondOnSuccess, isCiPipeline
...
+    if (environmentsLoading) return;
```
[the remark this answers is not in the document]
Comment (shivani170) [reply src/components/app/details/appConfig/AppConfig.tsx:378]: done

File: src/components/login/SSOLogin.tsx [hunk 7]
```diff
@@ -36,6 +45,29 @@ const ssoMap = {
...
+    switch (SSOName) {
+        case "Google": return <Google />
+        case "GitHub": return <GitHub />
+        case "Microsoft": return <Microsoft />
+        case "LDAP": return <LDAP />
+        case "OIDC": return <OIDC />
+        case "OpenShift": return <Openshift />
+    }
+}
+const SSOLoginTab: React.FC<{ handleSSOClick: (e) => void, checked: boolean, lastActiveSSO, value: string, SSOName: string }> = ({ handleSSOClick, checked, lastActiveSSO, value, SSOName }) => {
```
[the remark this answers is not in the document]
Comment (shivani170) [reply src/components/login/SSOLogin.tsx]: done

**Review by nishant-d — APPROVED** [review]

**Pull request merged**